src/main/scala/MulAddRecFN.scala

```diff
@@ -462,8 +462,8 @@ class MulAddRecFN_postMul(expWidth: Int, sigWidth: Int) extends Module
     val roundUp_sigY3 =
         (((sigX3 | roundMask)>>UInt(2)) + UInt(1))(sigWidth + 1, 0)
     val sigY3 =
-        Mux(~roundUp & ~roundEven, (sigX3 & ~roundMask)>>UInt(2),    UInt(0)) |
-        Mux(roundUp,          roundUp_sigY3,                         UInt(0)) |
+        Mux((~roundUp & ~roundEven).toBool, (sigX3 & ~roundMask)>>UInt(2),    UInt(0)) |
```

Comment:
Why do we need the Bool casts?  It's not inherently problematic; it just seems odd that roundUp/roundEven aren't already Bools.